src/nupic/engine/Spec.cpp

```diff
@@ -103,6 +103,16 @@ bool InputSpec::operator==(const InputSpec &o) const {
          dataType == o.dataType &&
          count == o.count && description == o.description;
 }
+std::ostream& operator<< (std::ostream& out, const InputSpec& self) {
+   out << "     description: " << self.description << "\n"
```

Comment:
> nitpick: use std::endl instead of \n for portability.

This is portable.  On windows, a '\n' is replaced with '\r\n' if written to a file.  The only difference between '\n' and std::endl is that the std::endl will also flush...sometimes.